armi/tests/test_plugins.py

```diff
@@ -74,5 +96,75 @@ def test_exposeInterfaces(self):
             self.assertIsInstance(kwargs, dict)
 
 
+class TestPluginsNeedDB(ArmiTestHelper):
+    """This test class provides test plugins a realistic ARMI DB file to act upon."""
+
+    @classmethod
+    def setUpClass(cls):
+        # We need to be in the TUTORIAL_DIR so that for `filesToMove` to work right.
+        os.chdir(TUTORIAL_DIR)
+
+        # Do this work in a temp dir, to avoid race conditions.
+        cls.dirChanger = TemporaryDirectoryChanger(filesToMove=TUTORIAL_FILES)
+        cls.dirChanger.__enter__()
+        runTutorialNotebook()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.dirChanger.__exit__(None, None, None)
+
+    def setUp(self):
+        cs = settings.Settings(f"{CASE_TITLE}.yaml")
+        newSettings = {}
+        newSettings["db"] = True
+        newSettings["detailAssemLocationsBOL"] = ["001-001"]
+        newSettings["loadStyle"] = "fromDB"
+        newSettings["reloadDBName"] = pathlib.Path(f"{CASE_TITLE}.h5").absolute()
+        newSettings["startNode"] = 1
+        cs = cs.modified(newSettings=newSettings)
+
+        self.td = TemporaryDirectoryChanger()
+        self.td.__enter__()
+
+        c = case.Case(cs)
+        case2 = c.clone(title="armiRun")
+        settings.setMasterCs(case2.cs)
+        self.o = case2.initializeOperator()
+        self.r = self.o.r
+
+        self.o.getInterface("main").interactBOL()
+
+    def tearDown(self):
+        self.o.getInterface("database").database.close()
+        self.r = None
+        self.o = None
+        self.td.__exit__(None, None, None)
+
+    def test_afterLoadDB(self):
+        r"""
+        Test the afterLoadDB hook works
+
+        Notes
+        -----
+        This test *assumes* the ``afterLoadDB()`` hook takes place at the end of
+        the ``DBI.loadState()`` method. We are not here to test that plugin hooks
+        work, but that this one in particular can affect the ``Reactor`` state.
+        """
+        # Get to the database state at the end of stack of time node 1.
+        dbi = self.o.getInterface("database")
+        dbi.loadState(0, 1)
+
+        bFuels = self.r.core.getBlocks(Flags.FUEL)
+        power = sum([b.p.power for b in bFuels])
+        self.assertEqual(power, 0.0)
+
+        plugin = TestPluginAfterLoadDB()
+        plugin.afterLoadDB(self.r.core, self.o.cs)
+
+        bFuels = self.r.core.getBlocks(Flags.FUEL)
+        powers = [b.p.power for b in bFuels]
+        self.assertEqual(sum(powers), len(powers))
+
```

Comment:
Should also check that the power total is not equal to zero. I think this would pass even if the powers on the blocks are still zero/unchanged.

Reply:
Added the check

Reply:
Thanks! just wanted to make sure the code was _actually_ doing something :)